src/utils/client.js

```diff
@@ -1,82 +1,79 @@
 const jose = require('node-jose');
 const axios = require('axios');
 const { v4 } = require('uuid');
+const queryString = require('query-string');
 const clientId = 'medmorph_backend';
 const keys = require('../keys/privateKey.json');
 
-class Client {
-  /**
-   *
-   * @param {*} jwks This clients private key set used for signing requests
-   */
-  constructor(jwks) {
-    this.jwks = jwks;
-  }
+/**
+ * Generate and return access token for the specified server. If tokenEndpoint
+ * is provided it will use that, otherwise it will query the smart configuration.
+ *
+ * @param {string} url  the fhir base url for the server to connect to
+ * @returns access token.
+ */
+async function connectToServer(url) {
+  // Generate the client_assertion jwt
+  const tokenEndpoint = await getTokenEndpoint(url);
+  const jwt = await generateJWT(clientId, tokenEndpoint);
 
-  /**
-   * Generate and return access token for the specified server
-   * @param {string} url  the fhir base url for the server to connect to
-   */
-  async connectToServer(url) {
-    const response = await axios.get(`${url}/.well-known/smart-configuration`);
-    const wellKnown = response.data.token_endpoint;
-    const jwt = await this.generateJWT(clientId, wellKnown, keys.kid);
+  const props = {
+    scope: 'system/*.read',
+    grant_type: 'client_credentials',
+    client_assertion_type: 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer',
+    client_assertion: jwt
+  };
 
-    return jwt;
-  }
-
-  /**
-   * Get the local keystore that contains the JWKs for this client
-   */
-  async getKeystore() {
-    if (this.keystore) {
-      return this.keystore;
-    }
-    if (this.jwks.keys) {
-      this.keystore = await jose.JWK.asKeyStore(this.jwks);
-    } else {
-      this.keystore = jose.JWK.createKeyStore();
-      this.keystore.add(this.jwks);
+  const headers = {
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+      'Accept-Encoding': 'gzip, deflate, br',
+      Connection: 'keep-alive',
+      Accept: '*/*'
     }
-    return this.keystore;
-  }
+  };
 
-  /**
-   * Get the key from the keystore for the kid provided.  If it is not there return
-   * the first key used for signing
-   * @param {string} kid - the kid of the key to lookup
-   */
-  async getKeyOrDefault(kid) {
-    const keystore = await this.getKeystore();
-    if (kid) {
-      return keystore.get(kid);
-    }
-    return keystore.all({ use: 'sig' });
-  }
+  // Get access token from auth server
+  const data = await axios
+    .post(tokenEndpoint, queryString.stringify(props), headers)
+    .then(response => response.data)
+    .catch(err => console.error(err));
+  return data.access_token;
```

Comment:
Just a note - here I updated this to return `data.access_token` but you also need the expiration. So I would make this return `data` and then in storage/servers.js get the access_token and expiration. Just as a heads up the expiration is returned in `data.expires_in` which is the seconds until expiration. So when you store it in the db you'll need to add the current date (and convert units since `Date.now()` returns milliseconds)

Reply:
Good call, updated in both places